specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json

```diff
@@ -4970,6 +4973,38 @@
         "encryptedCredential": {
           "type": "string",
           "description": "The encrypted credential used for authentication. Credentials are encrypted using the integration runtime credential manager. Type: string."
+        },
+        "servicePrincipalId": {
+          "type": "object",
```

Comment:
Please add "x-ms-format" in ADF swagger

Reply:
Added, and mirrored other connectors for when to add x-ms-format